mirrortest.pp

```diff
@@ -0,0 +1,44 @@
+  package { "lvm2":
+    ensure => "installed"
+  }
+
+
+  #-----------------------------------------------------------------
+  # LVM (Logical Volume Manager) mirrors in puppet by Martin Wangel.
+  # Tested locally, and with Amazon EBS volumes on EC2 instances.
+  #-----------------------------------------------------------------
+  $lvm_file_system="ext4"              # ext4 or xfs, xfs can not be resized by this version of this puppet module
+  $vgname="mmvg"                       # name of Volume Group
+  $lvname="mmlv"                       # name of Logical Volume
+  $devices=["/dev/sdb1","/dev/sdb2"]   # Physical Devices
+  $lv_fs_size="2G"                     # Size of Logical Volume
+
+  physical_volume { [$devices]:
+      ensure => present
+  }
+  volume_group { "${vgname}":
+      ensure => present,
+      physical_volumes => $devices,
+      require =>  [ Physical_volume[$devices] ]
+  } ->
+  logical_volume { "${lvname}":
+      ensure => present,
+      volume_group => $vgname,
+      size => $lv_fs_size,
+      mirror => 1,
+      mirrorlog => core,
+      region_size => 4,
+      alloc => normal,
+      no_sync => 1
+  } ->
+  filesystem { "/dev/${vgname}/${lvname}":
+      ensure => present,
+      fs_type => $lvm_file_system,
+  } ->
+  mount { "lvm-${vgname}-${lvname}":
+    name    => "/u",
+    ensure  => mounted,
+    device  => "/dev/${vgname}/${lvname}",
+    fstype  => $lvm_file_system,
+    options => "defaults"
+  }
```

Comment:
Would you like to add the example code to some other file or wiki first, or
shall we assume that people want to figure out the intricacies of mirroring
themselves?

   Martin Wangel


On Wednesday, February 4, 2015, Morgan Haskel <notifications@github.com>
wrote:

> In mirrortest.pp
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76#discussion_r24087142>
> :
>
> > +      mirrorlog => core,
> > +      region_size => 4,
> > +      alloc => normal,
> > +      no_sync => 1
> > +  } ->
> > +  filesystem { "/dev/${vgname}/${lvname}":
> > +      ensure => present,
> > +      fs_type => $lvm_file_system,
> > +  } ->
> > +  mount { "lvm-${vgname}-${lvname}":
> > +    name    => "/u",
> > +    ensure  => mounted,
> > +    device  => "/dev/${vgname}/${lvname}",
> > +    fstype  => $lvm_file_system,
> > +    options => "defaults"
> > +  }
>
> This file shouldn't be added.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/puppetlabs/puppetlabs-lvm/pull/76/files#r24087142>.
>


-- 
-----------------------
 Martin Wangel
 Iteego